--- gjs-1.30.0/Makefile.am.wiget	2011-12-27 16:13:38.169339140 +0100
+++ gjs-1.30.0/Makefile.am	2011-12-27 16:16:48.684343724 +0100
@@ -167,7 +167,8 @@ include Makefile-gjs-dbus.am
 include Makefile-modules.am
 include Makefile-examples.am
 
-pkglib_DATA = $(INTROSPECTION_GIRS) $(INTROSPECTION_GIRS:.gir=.typelib)
+typelibdir = $(pkglibdir)
+typelib_DATA = $(INTROSPECTION_GIRS) $(INTROSPECTION_GIRS:.gir=.typelib)
 
 ########################################################################
 bin_PROGRAMS += gjs-console
