--- gjs-1.38.1/Makefile.am.orig	2013-09-25 04:00:21.000000000 +0200
+++ gjs-1.38.1/Makefile.am	2013-09-28 23:17:07.442387448 +0200
@@ -188,7 +188,7 @@
 if ENABLE_SYSTEMTAP
 gjs/gjs.stp: gjs/gjs.stp.in Makefile
 	sed -e s,@EXPANDED_LIBDIR@,$(libdir), < $< > $@.tmp && mv $@.tmp $@
-tapsetdir   = $(DESTDIR)$(datadir)/systemtap/tapset
+tapsetdir   = $(datadir)/systemtap/tapset
 tapset_DATA = $(tapset_in_files:.stp.in=.stp)
 endif
 
